resetcloser.go

```diff
@@ -0,0 +1,29 @@
+package main
+
+import (
+	"io"
+)
+
+type ResetCloser struct {
```

Comment:
I don't really like it either, but it felt kind of Go-like, similar to `ReadCloser` for example.

What do you think of `AutoCloser`, and a method named `Release()`? Just like the now-deprecated (but not due to naming) `std::auto_ptr`.

Reply:
I find `ReadCloser` to be weird - I think it should be `ReaderCloser`. But perhaps they just combine `Read` and `Close` and stick `er` on the end (link most interfaces).

I thought of `AutoCloser` but there is nothing "auto" about it. You need to call `Close` to have it close.

Perhaps `ReleaseCloser` is the closest to `ReadCloser`. `MaybeCloser`? `IDontBloodyKnowCloser`? Up to you.

Reply:
What do you think about `CloseCanceller`? The method would obviously be called `Cancel()` instead of `Reset()` or `Release()`. I think this describes what it's doing - you can defer the closing of the resource, but you can also cancel the deferred close. Sounds englishy to me, and it also satisfies Go's XY-er naming style.